assets/queries/cloudFormation/ec2_network_Acl_overlap_ports/query.rego

```diff
@@ -0,0 +1,51 @@
+package Cx
+
+CxPolicy [result] {
+  document := input.document[i]
+  prop1 := document.Resources[name1]
+  prop2 := document.Resources[name2]
+
+  name1 != name2
+
+  prop1.Type == "AWS::EC2::NetworkAclEntry"
+  prop2.Type == "AWS::EC2::NetworkAclEntry"
+  
+  from1 := prop1.Properties.PortRange.From
+  to1 := prop1.Properties.PortRange.To
+  
+  from2 := prop2.Properties.PortRange.From
+  to2 := prop2.Properties.PortRange.To
+	
+  overlap([from1, to1], [from2, to2])
```

Comment:
we can simplify this check like this:

```
  range1 := numbers.range(from1, to1)
  range2 := numbers.range(from2, to2)
  check_overlap(range1, range2)
```
and then create a function called check overlap:
```
check_overlap(range1, range2) {
    some j,k
    range1[j] = range2[k]
}
```

Reply:
amazing comment. In fact, I did not remember to use this _numbers.range_ built-in function. This really works. I'm going to change it